### PR TITLE
feature: Allow selecting inference response content for automl generated models

### PR DESCRIPTION
**Pull request by pdasamzn**
*Description of changes:*
Allow selecting inference response content for Autopilot generated models

1. Added attach() method to AutoML to allow attaching an AutoML object
to an existing SageMaker AutoMLJob
2. Added create_model() method to AutoML class that returns a PipelineModel
that can be used to deploy or transform.
3. Added a utility function `validate_and_update_inference_response()` to AutoML 
class to update the inference container list based on the requested inference 
response keys. 

4. Updated and added unit tests and integ tests

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to any/all clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/automl/automl.py [hunk 1]
```diff
@@ -100,6 +101,63 @@ def fit(self, inputs=None, wait=True, logs=True, job_name=None):
...
+            sagemaker_session (sagemaker.session.Session): A SageMaker Session
+                object, used for SageMaker interactions (default: None). If not
+                specified, the one originally associated with the ``AutoML`` instance is used.:
+
+        Returns:
+
+        """
+        sagemaker_session = sagemaker_session or Session()
+
+        _auto_ml_job_desc = sagemaker_session.describe_auto_ml_job(job_name)
```
[the remark this answers is not in the document]
Comment (pdasamzn) [reply src/sagemaker/automl/automl.py]: fixed

File: src/sagemaker/automl/automl.py [hunk 2]
```diff
@@ -400,6 +451,114 @@ def _prepare_for_auto_ml_job(self, job_name=None):
         if self.output_path is None:
             self.output_path = "s3://{}/".format(self.sagemaker_session.default_bucket())
 
+    @classmethod
+    def _get_supported_inference_keys(cls, container, default=None):
```
[the remark this answers is not in the document]
Comment (pdasamzn) [reply src/sagemaker/automl/automl.py:458]: No need to be public either.

File: src/sagemaker/automl/automl.py [hunk 3]
```diff
@@ -187,49 +248,29 @@ def list_candidates(
 
         return self.sagemaker_session.list_candidates(**list_candidates_args)["Candidates"]
 
-    def deploy(
+    def create_model(
```
Comment (wiltonwu) [inline src/sagemaker/automl/automl.py:251]: Should this actually be private? I don't see a way to deploy an already created model here, maybe it exists outside of automl.py?

Comment (pdasamzn) [reply src/sagemaker/automl/automl.py:251]: no. I purposefully introduced the method create_model as public facing APIs. This does not really create the sagemaker model, but it allows to use the generated PipelineModel to perform batch transform as well.

see the test for an example usecase

**Pull request merged**